google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineTask.java

```diff
@@ -21,12 +21,20 @@
 /**
  * Encapsulates a task to be run on Google App Engine.
  */
-public interface AppEngineTask {
+public abstract class AppEngineTask {
```

Comment:
I'm OK with this for this instance. 

Generally I'm not a big fan of using abstract classes as interfaces because taken to the extreme, inheritors lose the ability to implement multiple interfaces (AKA abstract parent classes). I also tend to like really explicit interfaces that implementors have to think about in terms of the full contract.  One way to combine things is to have an interface + an abstract base class with some convenience implementations.